targets/REV3-Hardmon/main.cpp

```diff
@@ -4,24 +4,119 @@
+
+#define CAN_IRQ_POST_INTERRUPT_ROUTINE _tx_thread_context_restore()
+
 #include <core/io/CAN.hpp>
 #include <core/io/UART.hpp>
 #include <core/io/types/CANMessage.hpp>
 #include <core/manager.hpp>
-#include <core/utils/time.hpp>
-#include <core/utils/types/FixedQueue.hpp>
+#include <core/utils/log.hpp>
 
 #include <core/io/CANopen.hpp>
 
 #include <co_core.h>
 #include <co_if.h>
 #include <co_tmr.h>
 
+// rtos includes
+#include <core/rtos/BytePool.hpp>
+#include <core/rtos/Enums.hpp>
+#include <core/rtos/EventFlags.hpp>
+#include <core/rtos/Queue.hpp>
+#include <core/rtos/Semaphore.hpp>
+#include <core/rtos/Thread.hpp>
+#include <core/rtos/Timer.hpp>
+#include <core/rtos/tsio/ThreadUART.hpp>
+
 #include <Hardmon.hpp>
 
+namespace rtos = core::rtos;
 namespace io = core::io;
 namespace dev = core::dev;
 namespace time = core::time;
+namespace log = core::log;
+
+///////////////////////////////////////////////////////////////////////////////
+//RTOS GLOBAlS SETUP
+///////////////////////////////////////////////////////////////////////////////
+
+/// The size of the memory pool for the tx application
+#define TX_APP_MEM_POOL_SIZE 65536
+/// How often the model should take 1 step.
+#define MODEL_THREAD_TRIGGER_RATE S_TO_TICKS(5)
+
+// Model Thread Parameters
+#define MODEL_THREAD_STACK_SIZE 1024
+#define MODEL_THREAD_PRIORITY 1
+#define MODEL_THREAD_PREEMPT_THRESHOLD 1
+#define MODEL_THREAD_TIME_SLICE MS_TO_TICKS(20)
+#define MODEL_THREAD_AUTOSTART true
+
+// Powertrain CAN Receive Thread Parameters
+#define PT_CAN_RECEIVE_STACK_SIZE 1024
+#define PT_CAN_RECEIVE_PRIORITY 3
+#define PT_CAN_RECEIVE_PREEMPT_THRESHOLD 3
+#define PT_CAN_RECEIVE_TIME_SLICE MS_TO_TICKS(10)
+#define PT_CAN_RECEIVE_AUTOSTART true
+
+// Health Thread Parameters
+#define HEALTH_THREAD_STACK_SIZE 1024
+#define HEALTH_THREAD_PRIORITY 5
+#define HEALTH_THREAD_PREEMPT_THRESHOLD 5
+#define HEALTH_THREAD_TIME_SLICE MS_TO_TICKS(10)
+#define HEALTH_THREAD_AUTOSTART true
+
+// Accessory CAN Receive Thread Parameters
+#define ACC_CAN_RECEIVE_THREAD_STACK_SIZE 1024
+#define ACC_CAN_RECEIVE_THREAD_PRIORITY 4
+#define ACC_CAN_RECEIVE_THREAD_PREEMPT_THRESHOLD 4
+#define ACC_CAN_RECEIVE_THREAD_TIME_SLICE MS_TO_TICKS(10)
+#define ACC_CAN_RECEIVE_THREAD_AUTOSTART true
+
+// Thread Structs
+
+/**
+ * Struct that holds information needed for the model thread
+ */
+typedef struct {
+    vcu::Hardmon* hardmon;
+    rtos::EventFlags* triggerFlag;
+} modelThreadArgs_t;
+
+/**
+ * Struct that holds information needed for the powertrain CAN thread
+ */
+typedef struct {
+    vcu::Hardmon* hardmon;
+} powertrainCANReceiveThreadArgs_t;
+
+/**
+ * Struct that holds information needed for the health thread
+ */
+typedef struct {
+    vcu::Hardmon* hardmon;
+} healthThreadArgs_t;
+
+/**
+ * Struct that holds information needed for the accessory CANopen Thread
+ */
+typedef struct {
+    vcu::Hardmon* hardmon;
+    //todo: once threadsafe canopen is implemented, this should take an instance of that.
```

Comment:
This is might just me but every other file with a TODO is in all caps so just for consistency keep the same format here. 
```suggestion
    //TODO: once threadsafe canopen is implemented, this should take an instance of that.
```